httomo/runner/task_runner.py

```diff
@@ -334,8 +344,6 @@ def determine_max_slices(self, section: Section, slicing_dim: int):
             non_slice_dims_shape = output_dims
 
         if not has_gpu:
-            section.max_slices = min(
-                min(max_slices_methods), httomo.globals.MAX_CPU_SLICES
-            )
+            section.max_slices = min(min(max_slices_methods), MAX_CPU_SLICES)
```

Comment:
I know that this is just the formatter changing things and no logic has been altered, but it did make me wonder why there is another check of `if not has_gpu`.

What I mean is, as the code stands, why is there two different places that the max slices gets set for when the section contains all CPU methods? It feels like there should be only one place that this occurs, it feels confusing that it gets set once earlier, and then once more here? Or have I misunderstood what this part of the code is doing?

Reply:
removed in [39f5a8e](https://github.com/DiamondLightSource/httomo/pull/364/commits/39f5a8eb68e0fd611d0f2b9c5c3485f058261903)

Reply:
Sorry, to try and clarify, what I'm saying is that there's your additions that check `if not has_gpu` (starting at line 305):
```python
        if not has_gpu:
            section.max_slices = min(httomo.globals.MAX_CPU_SLICES, max_slices)
            return
```

but there's also something further down doing the exact same check and setting the max slices (starting at line 343):
```python
        if not has_gpu:
            section.max_slices = min(
                min(max_slices_methods), httomo.globals.MAX_CPU_SLICES
            )
```

What I'm asking is, why are there two checks for `if not has_gpu` to set `section.max_slices`?

Reply:
yes, you're right. Shouldn't be there. 
[e1254db](https://github.com/DiamondLightSource/httomo/pull/364/commits/e1254db3eb82f94185ffb885f05c19e2ac736207)